Sources/MongoSwift/MongoCollection.swift

```diff
@@ -103,6 +113,42 @@ public struct MongoCollection<T: Codable> {
         return self._client.operationExecutor.execute(operation, client: self._client, session: session)
     }
 
+    /**
+     * Renames this collection on the server. This method will return a handle to the renamed collection. The handle
+     * which this method is invoked on will continue to refer to the old collection, which will then be empty.
+     * The server will throw an error if the new name matches an existing collection unless the `dropTarget` option
+     * is set to true.
+     *
+     * Note: This method is not supported on sharded collections.
+     *
+     * - Parameters:
+     *   - to: A `String`, the new name for the collection
+     *   - options: Optional `RenameCollectionOptions` to use for the collection
+     *   - session: Optional `ClientSession` to use when executing this command
+     *
+     * - Returns:
+     *    An `EventLoopFuture<MongoCollection>` evaluating to a copy of the target `MongoCollection` with the new name.
+     *
+     *    If the future evaluates to an error, it is likely one of the following:
+     *    - `MongoError.CommandError` if an error occurs that prevents the command from executing.
+     *    - `MongoError.InvalidArgumentError` if the options passed in form an invalid combination.
+     *    - `MongoError.LogicError` if the provided session is inactive.
+     *    - `MongoError.LogicError` if this databases's parent client has already been closed.
+     *    - `EncodingError` if an error occurs while encoding the options to BSON.
+     */
+    public func renamed(
+        to: String,
```

Comment:
small nit: Swift has both "argument labels" and "parameter names". in this case, since you only provide one value `to`, "to" is used as _both_ the label and the name. 

(previously, when you had `_ to: String` here, `_` was the label (i.e. it had no label) and `to` was the name.)

in cases where the label is a preposition (like `to`) it's idiomatic to provide a separate name for the parameter, e.g. `to newName: String`. Then users get the benefit of calling it with the label (`rename(to: "abc")`) but in the method body it has a clearer name.